### PR TITLE
WT-7756 RTS to clear the HS flag of an update following a stable tombstone

### PR DESCRIPTION
**Pull request by kommiharibabu**

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -72,6 +72,10 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_UPDATE *first_upd,
      */
     if (first_upd != NULL) {
         F_CLR(first_upd, WT_UPDATE_HS);
+
+        /* Clear the history store flag of an update following a stable tombstone. */
+        if (first_upd->type == WT_UPDATE_TOMBSTONE && first_upd->next != NULL)
```
Comment (quchenhao) [inline src/txn/txn_rollback_to_stable.c:77]: Shall we walk through all the aborted tx in the middle? Though there shouldn't be any. May be we can assert it must be a full update restored from hs here.

**Comment by quchenhao** [comment]
Can we write a test to reproduce this?

File: src/txn/txn_rollback_to_stable.c (hunk 1 above)
Comment (hbokhari) [reply src/txn/txn_rollback_to_stable.c:77]: I am also curious in knowing if we are always certain that the next update (after tombstone) will never be an aborted update.

Comment (kommiharibabu) [reply src/txn/txn_rollback_to_stable.c]: It is possible that there can be aborted updates in between. Added additional checks to consider this scenario. The existing test is enhanced to consider this scenario.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by hbokhari — APPROVED** [review]

**Pull request merged**